stl/inc/algorithm

```diff
@@ -1017,6 +1015,34 @@ namespace ranges {
                 }
                 // If we get here, _Count > 1, initial elements do not match, and final elements do not match.
 
+                const auto _ProjectedPred = [&]<class _Ty1, class _Ty2>(_Ty1&& _Left, _Ty2&& _Right) {

```

Comment:
This might lead to code bloat as that lambda will have a unique name for each TU.

Also it is not really doing anything so we should simply go with the inoke sequence as everywhere else

Reply:
The purpose of this lambda is to avoid rewriting `_Trim_equal` and `_Trim_reversed` with the invoke sequence. Will it really cause code bloat? Shouldn't compiler inline it?

Reply:
@CaseyCarter and I talked about this while video reviewing - we can't exactly remember in what context we heard that "lambdas in STL algorithms are potentially bad for throughput" and it might be almost entirely mythical. I recommend not worrying about it here - the lambda depends on the type arguments, so it should result in the same codegen as a named helper function/struct. We can always refactor this in the future if it turns out to be a throughput issue.